server/worker/apple_mdm.go

```diff
@@ -64,30 +67,48 @@ func (a *AppleMDM) Run(ctx context.Context, argsJSON json.RawMessage) error {
 	case AppleMDMPostDEPEnrollmentTask:
 		err := a.runPostDEPEnrollment(ctx, args)
 		return ctxerr.Wrap(ctx, err, "running post Apple DEP enrollment task")
+
 	case AppleMDMPostManualEnrollmentTask:
 		err := a.runPostManualEnrollment(ctx, args)
 		return ctxerr.Wrap(ctx, err, "running post Apple manual enrollment task")
+
+	case AppleMDMPostDEPReleaseDeviceTask:
+		err := a.runPostDEPReleaseDevice(ctx, args)
+		return ctxerr.Wrap(ctx, err, "running post Apple DEP release device task")
+
 	default:
 		return ctxerr.Errorf(ctx, "unknown task: %v", args.Task)
 	}
 }
 
 func (a *AppleMDM) runPostManualEnrollment(ctx context.Context, args appleMDMArgs) error {
-	if err := a.installFleetd(ctx, args.HostUUID); err != nil {
+	if _, err := a.installFleetd(ctx, args.HostUUID); err != nil {
 		return ctxerr.Wrap(ctx, err, "installing post-enrollment packages")
 	}
 
 	return nil
 }
 
 func (a *AppleMDM) runPostDEPEnrollment(ctx context.Context, args appleMDMArgs) error {
-	if err := a.installFleetd(ctx, args.HostUUID); err != nil {
+	// TODO(mna): what happens if this function fails before everything got sent?
+	// On the next try, new commands will be enqueued for the same task (e.g. if
+	// it failed at the bootstrap package, install fleetd will be enqeueued
+	// again), will Apple MDM process both commands successfully or would the
+	// second one fail? Relevant for the device release.
```

Comment:
@roperzh maybe you know this? My hunch is that those are idempotent so if e.g. fleetd is already installed and the command is sent again, it would succeed again.

Reply:
it would! they're not idempotent, but I believe the commands we send won't produce errors either.

- All `InstallProfile` commands won't have any effect (because it'll be considered an 'edit' by the OS and nothing changed)
- All `InstallEnterpriseApplication` will trigger a second install. This _might_ be problematic for the bootstrap package, but I believe Noah's plan is to document that the bootstrap package must be "idempotent" (as much as an application can be)

I don't know if I'm missing other commands.

Reply:
> I believe the commands we send won't produce errors either.
> This might be problematic for the bootstrap package, but I believe Noah's plan is to document that the bootstrap package must be "idempotent" (as much as an application can be)

Awesome, thanks for the clarifications!

Reply:
> I don't know if I'm missing other commands.

The other potential one is `AccountConfiguration`, which I assume would be like an edit to the previous settings and would not result in an error.

Reply:
agreed, IIRC that's fine as well.